api/http/proxy/factory/docker/transport.go

```diff
@@ -381,9 +383,31 @@ func (transport *Transport) proxyTaskRequest(request *http.Request) (*http.Respo
 }
 
 func (transport *Transport) proxyBuildRequest(request *http.Request) (*http.Response, error) {
+	err := transport.updateDefaultGitBranch(request)
+	if err != nil {
+		return nil, err
+	}
 	return transport.interceptAndRewriteRequest(request, buildOperation)
 }
 
+func (transport *Transport) updateDefaultGitBranch(request *http.Request) error {
+	remote := request.URL.Query().Get("remote")
+	if strings.HasSuffix(remote, ".git") {
+		repositoryURL := remote[:len(remote)-4]
+		latestCommitID, err := transport.gitService.LatestCommitID(repositoryURL, "", "", "")
```

Comment:
Does it mean that with a `referenceName` being an empty string, LatestCommitID will literally pull a latest commit regardless of its reference? If that's true then Portainer won't necessarily pick the default branch and may result in building an image from an undesired reference. Is that line of thought correct or I might be missing something out in relation to the `HEAD`?

Could you please write a unit test for it to make sure it outputs desired urls in all cases.

Reply:
If "referenceName" is an empty string, LatestCommitID will return the latest commit ID of the default branch(ex. main).

Sure, will add some test cases for it.